query/src/sessions/session_settings.rs

```diff
@@ -444,6 +444,20 @@ impl Settings {
 
         Ok(())
     }
+
+    pub fn set_batch_settings(
+        &self,
+        settings: &HashMap<String, String>,
+        is_global: bool,
+    ) -> Result<()> {
+        for (k, v) in settings.iter() {
+            if self.has_setting(k.as_str()) {
```

Comment:
each has_setting， set_settings try get the lock
what about doing the batch upate with  one lock  cc @sundy-li

Reply:
Make sense.

Reply:
A trade-off between simple code and performance. But in this situation, it seems not be bottlelock? Generally speaking, there are no too many settings. In my opinion, I prefer keep this. What do you think?

Reply:
lgtm